app/controllers/media_objects_controller.rb

```diff
@@ -109,22 +109,25 @@ def new
 
   # POST /media_objects/avalon:1/add_to_playlist
   def add_to_playlist
-    @media_object = SpeedyAF::Proxy::MediaObject.find(params[:id])
+    if params[:post].present?
+      @media_object = params[:post][:masterfile_id].present? ? SpeedyAF::Proxy::MediaObject.find(params[:id]) : SpeedyAF::Proxy::MediaObject.find(params[:id], load_reflections: true)
+    end
```

Comment:
This guard seems a little odd since the rest of the action expects params[:post] to be present.  This might be a good place to use strong parameters to require the params are present.  The failing test(s) in the controller specs would need to have the required params added to the call.
```suggestion
    add_to_playlist_params = params.require(:post).permit(:masterfile_id, :playlist_id, :playlistitem_scope)
    @media_object = add_to_playlist_params[:masterfile_id].present? ? SpeedyAF::Proxy::MediaObject.find(params[:id]) : SpeedyAF::Proxy::MediaObject.find(params[:id], load_reflections: true)
```
Adding strong parameters might be out of scope for this PR, but it feels like the test(s) should have the parameters added either way so they reflect expected usage.